docs/proposals-accepted/202206-active-series-limiting-hashring.md

```diff
@@ -0,0 +1,207 @@
+
+<img src="../img/meta-monitoring-validator.png" alt="Meta-monitoring-based Validator" width="800"/>
+
+#### Pros:
+* Simpler as compared to other solutions and easier to implement
+* Lesser endpoint calls, so improved latency
+* Relies on "external to Thanos" system, and doesn’t increase load on Receive
+* Does not add much tenancy-based complexity to Thanos
+* No need to merge statistics across replicas, handled by meta-monitoring
+* Additional request-based rate limiting can be done within same component
+* In case, external meta-monitoring solution is down, can fall back to per-replica-tenant limits
+
+#### Cons:
+* Not very accurate
+  * We do not know exact state of each TSDB, only know view of meta-monitoring solution, which gets updated on every scrape
+  * We do not account for how much a remote write request will increase the number of active series, only infer that from query result after the fact
+* Data replication (quorum-based) will likely cause inaccuracies in HEAD stat metrics
+* Dependence on external system, so best effort availability
+
+### Receive Validator
+
+We can implement some new endpoints on Thanos Receive.
+
+Firstly, we can take advantage of the `api/v1/status/tsdb` endpoint that is exposed by [Prometheus TSDB](https://prometheus.io/docs/prometheus/latest/querying/api/#tsdb-stats) but is yet to be implemented in Thanos Receive ([in-review PR](https://github.com/thanos-io/thanos/pull/5402) which utilizes tenant headers to get local tenant TSDB stats in Receive).
+
+In its current WIP implementation, it can provide us stats for each local TSDB of a tenant which contains a measure of HEAD series (so active series). We can merge this to get the total number of HEAD or active series a tenant has.
+
+Furthermore, we also have each tenant’s [Appendable](https://pkg.go.dev/github.com/thanos-io/thanos/pkg/receive#Appendable) in multitsdb, which returns a Prometheus [storage.Appender](https://pkg.go.dev/github.com/prometheus/prometheus/storage#Appender), which can in turn give us a [storage.GetRef](https://pkg.go.dev/github.com/prometheus/prometheus/storage#GetRef.GetRef) interface. This helps us know if a TSDB has a cached reference for a particular set of labels in its HEAD.
+
+This GetRef interface returns a [SeriesRef](https://pkg.go.dev/github.com/prometheus/prometheus/storage#SeriesRef) when a set of labels is passed to it. If the SeriesRef is 0, it means that that set of labels is not cached, and any sample with that set of labels will generate a new active series. This data can also be fetched from a new endpoint like `api/v1/getrefmap` and merged across replicas.
+
+This approach would also add another mode to Receive apart from Router, Ingestor & RouterIngestor, called **“Validator”** which could actually be a Receive instance in front of Receive hashring. This is where we can get data from hashring Receivers and merge them to limit remote write requests.
```

Comment:
The receive router split proposal makes it possible to compose arbitrary trees of receivers - so we could leverage that instead of creating a new mode entirely.